subworkflows/local/bam_applybqsr_spark/main.nf

```diff
@@ -9,57 +9,39 @@ include { CRAM_MERGE_INDEX_SAMTOOLS } from '../cram_merge_index_samtools/main'
 
 workflow BAM_APPLYBQSR_SPARK {
     take:
-        cram          // channel: [mandatory] meta, cram, crai, recal
-        dict          // channel: [mandatory] dict
-        fasta         // channel: [mandatory] fasta
-        fasta_fai     // channel: [mandatory] fasta_fai
-        intervals     // channel: [mandatory] intervals, num_intervals
+    cram          // channel: [mandatory] meta, cram, crai, recal
+    dict          // channel: [mandatory] dict
+    fasta         // channel: [mandatory] fasta
+    fasta_fai     // channel: [mandatory] fasta_fai
+    intervals     // channel: [mandatory] intervals, num_intervals
 
     main:
-    ch_versions = Channel.empty()
+    versions = Channel.empty()
 
-    cram_intervals = cram.combine(intervals)
-        .map{ meta, cram, crai, recal, intervals, num_intervals ->
-
-            //If no interval file provided (0) then add empty list
-            intervals_new = num_intervals == 0 ? [] : intervals
-
-            [[
-                id:             meta.sample,
-                data_type:      meta.data_type,
-                num_intervals:  num_intervals,
-                patient:        meta.patient,
-                sample:         meta.sample,
-                sex:            meta.sex,
-                status:         meta.status,
-            ],
-            cram, crai, recal, intervals_new]
-        }
+    cram_intervals = cram.combine(intervals).map{ meta, cram, crai, recal, intervals, num_intervals ->
+        //If no interval file provided (0) then add empty list
+        [ meta.subMap('data_type', 'patient', 'sample', 'sex', 'status')
+            + [ id:meta.sample, num_intervals:num_intervals ],
+            cram, crai, recal, (num_intervals == 0 ? [] : intervals) ]
+    }
 
     // Run Applybqsr spark
     GATK4_APPLYBQSR_SPARK(cram_intervals, fasta, fasta_fai, dict)
 
     // STEP 4.5: MERGING AND INDEXING THE RECALIBRATED BAM FILES
     CRAM_MERGE_INDEX_SAMTOOLS(GATK4_APPLYBQSR_SPARK.out.cram, fasta, fasta_fai)
 
-    ch_cram_recal_out = CRAM_MERGE_INDEX_SAMTOOLS.out.cram_crai.map{ meta, cram, crai ->
-                             // remove no longer necessary fields to make sure joining can be done correctly: num_intervals
-                            [[
-                                id:         meta.id,
-                                data_type:  meta.data_type,
-                                patient:    meta.patient,
-                                sample:     meta.sample,
-                                sex:        meta.sex,
-                                status:     meta.status,
-                            ],
-                            cram, crai]
-                        }
+    cram_recal = CRAM_MERGE_INDEX_SAMTOOLS.out.cram_crai.map{ meta, cram, crai ->
+        // remove no longer necessary fields to make sure joining can be done correctly: num_intervals
+        [ meta.subMap('data_type', 'id', 'patient', 'sample', 'sex', 'status'), cram, crai ]
```

Comment:
since just things are removed, would just map.remove() work? Then we would avoid missing a field, I suppose

Reply:
Good point, @robsyme what do you say?